drupal/web/modules/custom/wunder_next/src/Commands/WunderNextCommands.php

```diff
@@ -15,115 +15,109 @@ class WunderNextCommands extends DrushCommands {
-  const API_USER_MAIL = 'next-api-user@domain.tld';
   const CONSUMER_NAME = 'next-drupal-consumer';
 
   /**
-   * Generates a user and a consumer to be used to view next-drupal previews.
-   *
-   * @param array $options
-   *   An associative array of options whose values come from cli, aliases,
-   *   config, etc.
-   *
-   * @option secret
-   *   The secret to assign to the consumer
-   * @option show_secrets
-   *   Choose if the secrets will be part of the output in plain text.
+   * Generates users and consumers needed for Next.js to speak to Drupal.
    *
-   * @usage wunder_next-commandName --secret=somesecret --show_secrets=TRUE
-   *   Use the --secret option to set the secret for the consumer.
-   *   If you leave it blank, the env var DRUPAL_CLIENT_SECRET
-   *   will be used, if not available a random value will be generated.
-   *   Set the show_secrets to TRUE to have the secrets be part of
-   *   the output in plain text.
-   *
-   * @command wunder_next:setup-user-and-consumer
+   * @command wunder_next:setup-users-and-consumers
    * @aliases wnsuac
    */
-  public function setupUserAndConsumer(array $options = [
-    'secret' => '',
-    'show_secrets' => FALSE,
-  ]) {
-
-    // For the secret, use the env var if defined, otherwise
-    // generate a random value:
-    $secret = $_ENV['DRUPAL_CLIENT_SECRET'] ?: base64_encode(random_bytes('20'));
-
-    // But if a value was provided as parameter to the command,
-    // use that instead:
-    $secret = $options['secret'] ?: $secret;
-
-    // The option to show the secrets in plain text.
-    $show_secrets = $options['show_secrets'];
-
-    // Create a new user with the required role:
-    $new_user = [
-      'name' => self::API_USER_NAME,
-      'pass' => '',
-      'mail' => self::API_USER_MAIL,
-      'access' => '0',
-      'status' => 1,
-      'roles' => [self::API_USER_ROLE],
+  public function setupUsersAndConsumers() {
+
+    // We want to have two separate consumers:
+    $consumers_to_create = [
+      // This first consumer is the usual consumer that next-drupal expects,
+      // and it is associated with a role that has elevated permissions to
+      // access unpublished content. This is used when generating previews for
+      // example.
+      [
+        'username' => self::API_USER_NAME,
+        'mail' => self::API_USER_MAIL,
+        'role' => self::API_USER_ROLE,
+        'secret' => $_ENV['DRUPAL_CLIENT_SECRET'],
+        'client_id' => $_ENV['DRUPAL_CLIENT_ID'],
+      ],
+      // This second consumer is associated with a role with fewer permissions,
+      // and it is used when the frontend user does not need access to
+      // unpublished content.
+      [
+        'username' => self::API_VIEWER_USER_NAME,
+        'mail' => self::API_VIEWER_USER_MAIL,
+        'role' => self::API_VIEWER_USER_ROLE,
+        'secret' => $_ENV['DRUPAL_CLIENT_VIEWER_SECRET'],
+        'additional_roles' => [self::FRONTEND_LOGIN_ROLE],
```

Comment:
this second consumer, for the "viewer" will also be used by the frontend login users, which now will not be able to see unpublished content